src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/HentaiVNParser.kt

```diff
@@ -33,23 +40,26 @@ class HentaiVNParser(context: MangaLoaderContext) : MangaParser(context, MangaSo
 	override suspend fun getDetails(manga: Manga): Manga = coroutineScope {
 		val chapterDeferred = async { fetchChapters(manga.url) }
 		val docs = webClient.httpGet(manga.url.toAbsoluteUrl(domain)).parseHtml()
-		val infoEl = docs.selectFirstOrThrow("div.container")
-			.selectFirstOrThrow("div.page-info")
+
+		val id = docs.location().substringAfterLast("/").substringBefore("-")
+
+		val infoEl = webClient.httpGet("/list-info-all-mobile.php?id_anime=$id".toAbsoluteUrl(domain)).parseHtml()
+
+		val genreUrl =  Regex(""""(list-info-theloai-mobile\.php?.+)"""").find(docs.toString())?.groupValues?.get(1)
+		val genre = webClient.httpGet("https://$domain/$genreUrl").parseHtml().select("a.tag")
+
+		val stateDoc = webClient.httpGet("/list-info-time-mobile.php?id_anime=$id".toAbsoluteUrl(domain)).parseHtml()
```

Comment:
These calls can be managed in parallel using `async()`

Reply:
Thanks for the correction, it was added.

Reply:
Uh, calling `await` right after `async` has no sense, right?)

I'll fix it by my own

Reply:
Sorry, I thought it worked differently, but thanks for changing it.